File: Algorithms/Implementation/The Time in Words.php
<?php

/*
 * Complete the 'timeInWords' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts following parameters:
 *  1. INTEGER h
 *  2. INTEGER m
 */

function timeInWords($h, $m) {
    # Write your code here
    $numbers = [
        1 => "one",
        2 => "two",
        3 => "three",
        4 => "four",
        5 => "five",
        6 => "six",
        7 => "seven",
        8 => "eight",
        9 => "nine",
        10 => "ten",
        11 => "eleven",
        12 => "twelve",
        13 => "thirteen",
        14 => "fourteen",
        15 => "fifteen",
        16 => "sixteen",
        17 => "seventeen",
        18 => "eighteen",
        19 => "nineteen",
        20 => "twenty",
        21 => "twenty one",
        22 => "twenty two",
        23 => "twenty three",
        24 => "twenty four",
        25 => "twenty five",
        26 => "twenty six",
        27 => "twenty seven",
        28 => "twenty eight",
        29 => "twenty nine"
    ];

    if ($m == 0) {
        return $numbers[$h] . " o' clock";
    }
    elseif ($m == 15) {
        return "quarter past " . $numbers[$h];
    }
    elseif ($m == 30) {
        return "half past " . $numbers[$h];
    }
    elseif ($m == 45) {
        return "quarter to " . $numbers[($h % 12) + 1];
    }
    elseif ($m < 30) {
        $minuteWord = ($m == 1) ? "minute" : "minutes";
        return $numbers[$m] . " " . $minuteWord . " past " . $numbers[$h];
    }
    else {
        $minuteWord = ((60 - $m) == 1) ? "minute" : "minutes";
        return $numbers[60 - $m] . " " . $minuteWord . " to " . $numbers[($h % 12) + 1];
    }
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$h = intval(trim(fgets(STDIN)));
$m = intval(trim(fgets(STDIN)));
$result = timeInWords($h, $m);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Implementation/The Time in Words.py
import os

#
# Complete the 'timeInWords' function below.
#
# The function is expected to return a STRING.
# The function accepts following parameters:
#  1. INTEGER h
#  2. INTEGER m
#

def timeInWords(h, m):
    # Write your code here
    numbers = [
        "one",
        "two",
        "three",
        "four",
        "five",
        "six",
        "seven",
        "eight",
        "nine",
        "ten",
        "eleven",
        "twelve",
        "thirteen",
        "fourteen",
        "fifteen",
        "sixteen",
        "seventeen",
        "eighteen",
        "nineteen",
        "twenty",
        "twenty one",
        "twenty two",
        "twenty three",
        "twenty four",
        "twenty five",
        "twenty six",
        "twenty seven",
        "twenty eight",
        "twenty nine",
    ]
    if m == 0:
        return f"{numbers[h-1]} o' clock"
    elif m == 15:
        return f"quarter past {numbers[h-1]}"
    elif m == 30:
        return f"half past {numbers[h-1]}"
    elif m == 45:
        return f"quarter to {numbers[h % 12]}"
    elif m < 30:
        minute_word = "minute" if m == 1 else "minutes"
        return f"{numbers[m-1]} {minute_word} past {numbers[h-1]}"
    else:
        minute_word = "minute" if 60 - m == 1 else "minutes"
        return f"{numbers[59 - m]} {minute_word} to {numbers[h % 12]}"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    h = int(input().strip())
    m = int(input().strip())
    result = timeInWords(h, m)

    fptr.write(result + '\n')
    fptr.close()
